src/widgetastic_patternfly/__init__.py

```diff
@@ -2274,3 +2282,219 @@ class AggregateStatusMiniCard(AggregateStatusCard):
         'and h2[contains(@class, "card-pf-title")]'
         '//span[normalize-space(following::text())={@name|quote}]]'
     )
+
+
+class SparkLineChart(Widget, ClickableMixin):
+    """Represents the Spark Line Chart from Patternfly (Data Visualization).
+
+    Args:
+        id: id of SparkLineChart
+        locator: If id not applies, you can also supply a full locator
+
+    .. code-block:: python
+        spark_line_chart = SparkLineChart(id="sparklineChart")
+    """
+    ROOT = ParametrizedLocator("{@locator}")
+    BASE_LOCATOR = ".//div[@id={}]"
```

Comment:
Please include 'chart-pf-sparkline' class in the default/BASE_LOCATOR to be more explicit here.  The Patternfly spec includes that class. Multiple div's on the page could have the same ID but different classes. (may be a poor page design, but its possible).

Reply:
@mshriver I agreed it will provide `flexibility` and also agreed with `may be a poor page design` possibility. I tried changes locally and  found some observations.

If we add `class` like `chart-pf-sparkline` in `BaseLocator` as  per standard design. most of user need to inherit and create own `BaseLocator`. Above condition they can handle by:
1. We are  providing `locator` as one `argument`. Users can use `locator`; if they found such thing in their page design.
2. If they like to add check  for `class` and don't  want to provide locator for specific graph. they can go with inheritance and overwrite `BASE_LOCATOR` as per their own requirements like....
```
BASE_LOCATOR = ".//div[@id={id} and contains(@class, 'chart-miq-spark')]"
```

What do you think? we should  go with more global approach so user can use widget directly and customize only if they needs.

Reply:
Thanks Nikhil - I think that the _default_ behavior of the widget needs to model the patternfly reference implementation, and be as precise as possible in its xpath locators.

For that reason, I would prefer to have the patternfly reference class in the base locator XPATH. Just as you said, if I want to completely override the locator because my project uses a significantly different base element, I can do that directly.  Otherwise, if just the class changes, then subclassing as below is very simple, and explicitly shows how the markup changes for the project where this is used.

```
class MyProjectSparkLine(SparkLineChart):
    BASE_LOCATOR = '//div[@id={} and contains(@class, 'myproject-spark-class')]'
```